frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java

```diff
@@ -363,4 +368,50 @@ public static VerificationState getVerificationState(VulnerabilityDTO vul){
                                 .findFirst()
                                 .orElse("")));
     }
+
+    public static Map<String, Integer> getServeListMapFromRequest(ResourceRequest request) {
+        HttpServletRequest httpReq = PortalUtil.getOriginalServletRequest(PortalUtil.getHttpServletRequest(request));
+        Map<String, Integer> serveListMap = new HashMap<>();
+
+        int listDisplayAmount = 25;
+        int startIdx = 0;
+        int sortColumnIdx = 1;
+        int sortDir = 0;
+
+        String pageNo = httpReq.getParameter(DATATABLE_DISPLAY_START);
+        String pageSize = httpReq.getParameter(DATATABLE_DISPLAY_LENGTH);
+        String colIndex = httpReq.getParameter(DATATABLE_SORT_COLUMN);
+        String sortDirection = httpReq.getParameter(DATATABLE_SORT_DIRECTION);
+
+        if (pageNo != null) {
+            startIdx = Integer.parseInt(pageNo);
+            if (startIdx < 0) {
+                startIdx = 0;
+            }
+        }
+
+        if (pageSize != null) {
+            listDisplayAmount = Integer.parseInt(pageSize);
+            if (listDisplayAmount < -1 || listDisplayAmount > 100) {
+                listDisplayAmount = 25;
+            }
+        }
+
+        if (colIndex != null) {
+            sortColumnIdx = Integer.parseInt(colIndex);
+            if (sortColumnIdx < 0 || sortColumnIdx > 4) {
+                sortColumnIdx = 1;
+            }
+        }
+
+        if (sortDirection != null && !sortDirection.equals(SORT_ASCENDING)) {
+            sortDir = 1;
+        }
+
+        serveListMap.put(DATATABLE_DISPLAY_START, startIdx);
```

Comment:
Why not introduce a custom object to store that information instead of a generic map with special keys?

Reply:
created `PaginationParameters` and `PaginationParser`